RELEASING.md

```diff
@@ -0,0 +1,94 @@
+# AWS OpenTelemetry Java Instrumentation Release Process
+
+## Starting the Release
+
+Open the release build workflow in your browser [here](https://github.com/aws-observability/aws-opentelemetry-java-instrumentation/actions?query=workflow%3A%22Release+Build%22).
+
+You will see a button that says "Run workflow". Press the button, enter the version number you want
+to release in the input field that pops up, and then press "Run workflow".
+
+This triggers the release process, which builds the artifacts, updates the README with the new
+version numbers, commits the change to the README, publishes the artifacts, creates and pushes
+a git tag with the version number, and drafts a release with the agent artifact attached.
+
+## Announcement
+   
+Once the GitHub workflow completes, go to Github [release
+page](https://github.com/aws-observability/aws-java-otel-instrumentation/releases), and find the
+generated draft release to write release notes about the new release and add release notes.
+
+You can use `git log upstream/v$MAJOR.$((MINOR-1)).x..upstream/v$MAJOR.$MINOR.x --graph --first-parent`
+or the Github [compare tool](https://github.com/open-telemetry/opentelemetry-java/compare/)
+to view a summary of all commits since last release as a reference.
+
+## Patch Release
```

Comment:
I'm confused why all this cherry-picking is necessary for a patch release and why it's not the same as any other release. If we're deciding to make a release, why can't we just input the new version with only the patch version incremented and use the original workflow?

Reply:
We might have done a lot of development on HEAD since the previous minor release. If we find a critical bug in a minor release within a day or something, we'd probably skip it and just release HEAD directly. But if we find it in a couple of weeks, we don't want to release new features, or significant potentially not-well-baked changes, in a patch release and should cherry-pick into a branch from the previous minor release tag to make sure they're not included. Depending on the nature of a hotfix and timeline, we could make a call on whether to just go straight to the next minor from HEAD, or release a patch version of the previous.

Reply:
Ok I see, makes sense to have that flexibility. Thanks for explaining!